rero_ils/modules/stats/api.py

```diff
@@ -0,0 +1,277 @@
+            'term', organisation__pid=organisation_pid).count()
+
+    def number_of_librarians(self, library_pid):
+        """Number of users with a librarian role.
+
+        point in time
+        :param library_pid: string - the library to filter with
+        :return: the number of matched librarians
+        :rtype: integer
+        """
+        return PatronsSearch()\
+            .filter('term', roles='librarian')\
+            .filter('term', libraries__pid=library_pid)\
+            .count()
+
+    def number_of_active_patrons(self, library_pid):
+        """Number of patrons who did a transaction in a the past 365 days.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched active patrons
+        :rtype: integer
+        """
+        ptrns = set()
+        _to = self.now.format(fmt='YYYY-MM-DDT23:59:59')
+        _from = (self.now - relativedelta(months=12))\
+            .format(fmt='YYYY-MM-DDT00:00:00')
+        date_range = {'gte': _from, 'lte': _to}
+        for res in RecordsSearch(index=LoanOperationLog.index_name)\
+                .filter('range', date=date_range)\
+                .filter('terms', loan__trigger=['checkout', 'extend'])\
+                .filter('term', loan__item__library_pid=library_pid)\
+                .source(['loan'])\
+                .scan():
+            ptrns.add(res.loan.patron.hashed_pid)
+
+        return len(ptrns)
+
+    def number_of_order_lines(self, library_pid):
+        """Number of order lines created during the specified timeframe.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched order lines
+        :rtype: integer
+        """
+        return AcqOrderLinesSearch()\
+            .filter('range', _created=self.date_range)\
+            .filter('term', library__pid=library_pid).count()
+
+    def number_of_checkouts(self, library_pid):
+        """Number of checkouts of a library during the specified timeframe.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched checkouts
+        :rtype: integer
+        """
+        return RecordsSearch(index=LoanOperationLog.index_name)\
+            .filter('range', date=self.date_range)\
+            .filter('term', loan__trigger='checkout')\
+            .filter('term', loan__item__library_pid=library_pid)\
+            .count()
+
+    def number_of_renewals(self, library_pid):
+        """Number of renewals of a library during the specified timeframe.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched renewals
+        :rtype: integer
+        """
+        return RecordsSearch(index=LoanOperationLog.index_name)\
+            .filter('range', date=self.date_range)\
+            .filter('term', loan__trigger='extend')\
+            .filter('term', loan__item__library_pid=library_pid)\
+            .count()
```

Comment:
maybe useful to have a generic function `number_of_circ_operation(self, ibrary_pid, operation_name)`

Reply:
Good idea.